main.py

```diff
@@ -33,16 +34,33 @@
 LEXERS = sorted(get_all_lexers(), key=lambda l: l[0].lower())
 
 
-def highlight_paste(paste):
+def highlight_paste(paste, hl_lines):
     '''Use pygments to syntax highlight a paste, returns by the way the CSS'''
     lexer = get_lexer_by_name(paste.hl_alias)
-    formatter = HtmlFormatter(linenos=True, cssclass='source')
+    formatter = HtmlFormatter(linenos=True, cssclass='source', hl_lines=hl_lines)
     return (
         highlight(paste.content, lexer, formatter),
         formatter.get_style_defs('.source')
     )
 
 
+def parse_hl(hl):
+    if hl is None:
+        return []
+
+    if not re.match(r"\d+|(\d+-\d+)( \d+|(\d+-\d+))*", hl):
```

Comment:
Can you compile the regex once in the file ? Cf https://docs.python.org/3/library/re.html#re.compile